### PR TITLE
Appt 141 - Auth and basic layout

### PR DESCRIPTION
**Pull request by vicr1**

### CODE REVIEW

File: src/api/Nhs.Appointments.Api/Functions/GetUserProfileFunction.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+using System.Threading.Tasks;
+
+namespace Nhs.Appointments.Api.Functions
+{
+    public class GetUserProfileFunction(IValidator<EmptyRequest> validator, IUserContextProvider userContextProvider, ILogger<GetSitesForUserFunction> logger)
+    : BaseApiFunction<EmptyRequest, UserProfile>(validator, userContextProvider, logger)
+    {
+
+        [OpenApiOperation(operationId: "GetUserProfile", tags: new[] { "Utility" }, Summary = "Gets information about the signed in user")]
+        [OpenApiRequestBody("text/json", typeof(SetBookingStatusRequest))]
```
Comment (sambiramairelogic) [inline src/api/Nhs.Appointments.Api/Functions/GetUserProfileFunction.cs:23]: Looks like this request type might be a copypasta mistake :)

File: src/api/Nhs.Appointments.Api/Functions/GetUserProfileFunction.cs
Comment (kicr1) [inline src/api/Nhs.Appointments.Api/Functions/GetUserProfileFunction.cs:1]: If we are not using the GetSitesForUsers endpoint anymore would it be worth removing it now?

File: src/api/Nhs.Appointments.Api/Functions/GetUserProfileFunction.cs (hunk 1 above)
Comment (vicr1) [reply src/api/Nhs.Appointments.Api/Functions/GetUserProfileFunction.cs:23]: Indeed! I have resolved this in the very next PR which includes changes to this endpoint

File: src/api/Nhs.Appointments.Api/Functions/GetUserProfileFunction.cs
Comment (vicr1) [reply src/api/Nhs.Appointments.Api/Functions/GetUserProfileFunction.cs:1]: I have removed it in the next PR when getting site becomes part of the user profile endpoint

**Pull request merged**